### ISSUE TITLE
cannot call method "helpers" of undefined when moving meteor code to `src`

### ISSUE THREAD

**Issue by cmoscardi** [Sanjo/meteor-jasmine#244]
Hi, so we moved our source code into a `src` directory (previously `client`, `server`, and `both` lived in root). Upon moving it, jasmine is unhappy about client-side basic packaging? In particular `Template.helpers` seems to be undefined.

My feeling is that it has to do with mocks, but I find it odd that specifically switching the directory where code is loaded from causes issues. (this may be because I'm new to meteor).

Here's the basic log output:
```
W20150702-04:00:52.218(0)? (STDERR) connections property is deprecated. Use getConnections() method
W20150702-04:00:54.717(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/announcements/announcement.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.740(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/announcements/announcementCreate.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.746(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/home/home.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.752(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/layout/layout.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.758(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/nav/nav.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.764(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/users/userCreate.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.772(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/users/userEdit.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.785(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/users/userProfile.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.798(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/users/userSearch.js" has syntax errors. [TypeError: Cannot call method 'helpers' of undefined]
W20150702-04:00:54.810(0)? (STDERR) [sanjo:jasmine]: The file "/var/www/src/client/templates/users/username.js
```

Here's some more info:
-[verbose output](https://gist.github.com/cmoscardi/c1642d1e038ddb0efe15).
-[versions file](https://gist.github.com/cmoscardi/af925fa7cd5e459eb1bb).

Let me know if I can provide any other information.

**Comment by Sanjo** [comment]
Yeah, seems to be a bug where the server unit mode tries to load client only files. Can you move your server unit mode tests to the server integration mode? I will remove the server unit mode in the next release.

**Comment by ramijarrar** [comment]
@Sanjo Are you planning to move the unit testing modes into a separate package? or depreciate the functionality altogether?

**Comment by Sanjo** [comment]
My plan is to remove them altogether. If needed, I can improve the remaining server and client mode for unit testing.

I think with ES6 modules support in Meteor, unit testing will be better. And I will then work on a better unit testing solution for Meteor.

So for now you can use the integration modes for unit testing. If you use packages you can also write unit tests for packages with Jasmine (this also uses the integration modes).

**Comment by ramijarrar** [comment]
I'm interested in the reasoning behind this move considering the amount of effort that went into building them after the depreciation of jasmine-unit. Just to clarify, will this mean all jasmine tests will run in the meteor context?

**Comment by Sanjo** [comment]
I think that more work would be necessary to make the unit modes really good. For me the value of this work is currently not high enough to do it. Also with my current approaches for the unit modes I don't leverage the Meteor Core. I think the right user journey for unit testing is a tight integration into Meteor (that will come in the next 6-12 months I think. And I will help.).

Also I think in the current state most of the people are better off just using the integration modes for all unit and integration tests they write. The unit modes have a lot more rough edges. By removing the unit modes I can make the user journey better.

> Just to clarify, this will mean all jasmine tests will now run in the meteor context?

Yes. But you can still mock, stub and spy on things. This works on a test by test basis. spyOn does this already.

Let me know if you have any other concerns or problems with the integration modes. Feedback is welcomed.

**Comment by Sanjo** [comment]
And also no work is thrown away. I will use all the experience I gained for the future work. Karma may will be the client test runner for unit tests is the future (would make sense). Also the autostubbing will be used in some form. I just have to make it work for client also. Anyway the autostubbing work came from Jest ;-)
